api/comment_threads.rb

```diff
@@ -27,9 +27,12 @@
     error 404, [t(:requested_object_not_found)].to_json
   end
 
-  if params["user_id"] and bool_mark_as_read
+  if params["user_id"]
     user = User.only([:id, :username, :read_states]).find_by(external_id: params["user_id"])
-    user.mark_as_read(thread) if user
+  end
+
+  if user and bool_mark_as_read
+    user.mark_as_read(thread)
```

Comment:
"P.S: I do not like the fact that we keep many logic as is in Forums ruby app while we are not using or changed the logic at platform mobile API side; just to keep Web working. This is making Forums app messy with almost every new change."

@wajeeha-khalid I agree.  I'm CCing @robrap and @scottrish so they are aware of this issue as well.